plugins/UsersManager/Controller.php

```diff
@@ -381,30 +381,33 @@ public function addNewToken()
         if (!empty($_POST['description'])) {
             Nonce::checkNonce(self::NONCE_ADD_AUTH_TOKEN);
 
-            $description = Common::getRequestVar('description', '', 'string');
+            $description = \Piwik\Request::fromRequest()->getStringParameter('description', '');
```

Comment:
Good point :+1: 
I've added a `strip_tags()` call.

Might be good to have some documented best practices for this so we don't reinvent the wheel each time. I guess https://github.com/matomo-org/developer-documentation/issues/669 will cover that.

Reply:
`strip_tags` might not be the best option to use. In general we should ensure to escape data, that can be provided by the user, whenever it is used in output. So in this case that's basically only the case, when listing the available tokens.
In twig all variables are automatically escaped with the `html` filter, so unless the variable isn't used `|raw` there should be nothing to do. But we should always double check that.
It might also be good to maybe adjust the UI tests, so one token in the list contains a description that was created with possible xss content. You could e.g. adjust this test: https://github.com/matomo-org/matomo/blob/5.x-dev/plugins/UsersManager/tests/UI/UserSettings_spec.js#L50
And add an additional showing the overview afterwards, to check the content is displayed correctly.